<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PyTorchGui</class>
 <widget class="QMainWindow" name="PyTorchGui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>851</width>
    <height>637</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PyTorch GUI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QTabWidget" name="tabWidget">
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="data_tab">
         <attribute name="title">
          <string>Data</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="2" column="1">
             <widget class="QPushButton" name="openDataFileButton">
              <property name="text">
               <string>Open data</string>
              </property>
              <property name="iconSize">
               <size>
                <width>8</width>
                <height>8</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QComboBox" name="dataTypePicker">
              <item>
               <property name="text">
                <string>Select the type of Data</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>CIFAR10</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>CIFAR100</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>MNIST</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QGraphicsView" name="graphicsView"/>
            </item>
            <item row="6" column="0">
             <widget class="QTableWidget" name="dataTable">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="rowCount">
               <number>0</number>
              </property>
              <property name="columnCount">
               <number>3</number>
              </property>
              <attribute name="horizontalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <attribute name="horizontalHeaderCascadingSectionResizes">
               <bool>false</bool>
              </attribute>
              <attribute name="horizontalHeaderHighlightSections">
               <bool>true</bool>
              </attribute>
              <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
               <bool>true</bool>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <attribute name="verticalHeaderCascadingSectionResizes">
               <bool>false</bool>
              </attribute>
              <column/>
              <column/>
              <column/>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QPushButton" name="clearDataButton">
              <property name="text">
               <string>Clear data</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string/>
              </property>
              <widget class="QRadioButton" name="trainingSetButton">
               <property name="geometry">
                <rect>
                 <x>140</x>
                 <y>5</y>
                 <width>100</width>
                 <height>21</height>
                </rect>
               </property>
               <property name="text">
                <string>Training set</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
              <widget class="QRadioButton" name="testingSetButton">
               <property name="geometry">
                <rect>
                 <x>260</x>
                 <y>0</y>
                 <width>96</width>
                 <height>31</height>
                </rect>
               </property>
               <property name="text">
                <string>Testing set</string>
               </property>
              </widget>
              <widget class="QLabel" name="label">
               <property name="geometry">
                <rect>
                 <x>16</x>
                 <y>4</y>
                 <width>111</width>
                 <height>21</height>
                </rect>
               </property>
               <property name="text">
                <string>Import data as :</string>
               </property>
              </widget>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="network_tab">
         <attribute name="title">
          <string>Network</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="defineNetworkLabel">
            <property name="text">
             <string>Define a network</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2"/>
          </item>
          <item>
           <widget class="QGraphicsView" name="netGFView"/>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QComboBox" name="layerPicker"/>
            </item>
            <item>
             <widget class="QPushButton" name="addLayerButton">
              <property name="text">
               <string>Add a layer</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QComboBox" name="activationFunctionPicker">
              <property name="currentIndex">
               <number>-1</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="addActivationFunctionButton">
              <property name="text">
               <string>Add an activation function</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QComboBox" name="addedLayerPicker"/>
            </item>
            <item>
             <widget class="QPushButton" name="editLayerButton">
              <property name="text">
               <string>Edit layer</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="setFirstLayerButton">
              <property name="text">
               <string>Set as first layer</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QComboBox" name="lossFunctionPicker">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <item>
               <property name="text">
                <string>Select a loss function</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="setLossFunctionButton">
              <property name="text">
               <string>Confirm</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="optimizerFunctionPicker">
              <item>
               <property name="text">
                <string>Select optimizer function</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="setOptimizerFunctionButton">
              <property name="text">
               <string>Confirm</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QPushButton" name="loadNetworkButton">
              <property name="text">
               <string>Load</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="saveNetworkButton">
              <property name="text">
               <string>Save</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="train_tab">
         <attribute name="title">
          <string>Train</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="trainingSetLabel">
              <property name="text">
               <string>Training set</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="trainingSetPicker">
              <property name="accessibleName">
               <string/>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="epochNumberlabel">
              <property name="text">
               <string>Number of epochs</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="epochNumberBox">
              <property name="maximum">
               <number>1000000000</number>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QPushButton" name="startTrainButton">
              <property name="text">
               <string>Start Train</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="batchSizeLabel">
              <property name="text">
               <string>Batch size</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="batchSizeBox">
              <property name="maximum">
               <number>100000</number>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="numWorkersLabel">
              <property name="text">
               <string>Workers number</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QCheckBox" name="shufflingCheckBox">
              <property name="text">
               <string>Shuffling</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="numWorkersBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>64</number>
              </property>
              <property name="value">
               <number>2</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="eval_tab">
         <attribute name="title">
          <string>Evaluation</string>
         </attribute>
         <widget class="QLabel" name="testSetLabel">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>71</width>
            <height>16</height>
           </rect>
          </property>
          <property name="text">
           <string>Testing set</string>
          </property>
         </widget>
         <widget class="QComboBox" name="testSetPicker">
          <property name="geometry">
           <rect>
            <x>110</x>
            <y>30</y>
            <width>76</width>
            <height>28</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="launchTestButton">
          <property name="geometry">
           <rect>
            <x>370</x>
            <y>490</y>
            <width>89</width>
            <height>30</height>
           </rect>
          </property>
          <property name="text">
           <string>Launch test</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>851</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Ima&amp;ge</string>
    </property>
    <addaction name="actionShow_landmarks"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
  </widget>
  <action name="actionShow_landmarks">
   <property name="text">
    <string>&amp;Show landmarks</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New Scratch</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>PyTorchGui</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
